### DISCUSSION TITLE
BE 1.2.4.1启动不了，报"Unrecognized VM option ':-CriticalJNINatives'，centos ,open jdk 1.8.0-372

### DISCUSSION THREAD

**Discussion by ahboer** [apache/doris#19334] [A - General / Q&A]
如题，
![image](https://user-images.githubusercontent.com/23441968/236605253-2b0c9ea9-4b92-4d8e-9a66-cec71d2424ea.png)

**Comment by ahboer** [comment]
喔，是统信1050e 基于x86-64的操作系统

**Reply by yagagagaga** [reply]
Can you execute `java -XX:+PrintFlagsFinal version | grep CriticalJNINatives` and show me the result?

**Reply by xavierzhao** [reply]
![image](https://github.com/apache/doris/assets/1573662/678adbf3-6b77-4420-9f88-4017a59ee0e3)

@yagagagaga

**Comment by ahboer** [comment]
***@***.***
 
发件人： yagagagaga
发送时间： 2023-05-06 16:12
收件人： apache/doris
抄送： ahboer; Author
主题： Re: [apache/doris] BE 1.2.4.1启动不了，报"Unrecognized VM option ':-CriticalJNINatives'，centos ,open jdk 1.8.0-372 (Discussion #19334)
Can you execute java -XX:+PrintFlagsFinal version | grep CriticalJNINatives and show me the result?

**Comment by ahboer** [comment]
I guess the following pic is what you need.



***@***.***
 
发件人： yagagagaga
发送时间： 2023-05-06 16:12
收件人： apache/doris
抄送： ahboer; Author
主题： Re: [apache/doris] BE 1.2.4.1启动不了，报"Unrecognized VM option ':-CriticalJNINatives'，centos ,open jdk 1.8.0-372 (Discussion #19334)
Can you execute java -XX:+PrintFlagsFinal version | grep CriticalJNINatives and show me the result?

**Comment by yagagagaga** [comment]
`CriticalJNiNatives` is one of the startup option in the JVM, which usually defaults `true`. Due to Doris using some Java Native Interfaces, it needs to be turned off(`-CriticalJNiNatives` means turned off and `+CriticalJNiNatives` means turned on). In the `start_be.sh` script, there is a line like `COMMON_OPTS="-Dsun.java.command=DorisBE -XX:-CriticalJNINatives"`. Generally, script execution will not cause problems, but according to your description, it seems that something wrong when you execute `start_be.sh`. 

Perhaps you can try executing `bash start_be.sh` or change to a different version of JDK, such as `OpenJDK Runtime Environment (build 1.8.0_372-b07)`

**Comment by ahboer** [comment]
Got it, the JVM_OPTS in the be.conf  has a double colon from the original package .I delete one colon,and the BE is running.
Thanks a lot!




***@***.***
 
发件人： yagagagaga
发送时间： 2023-05-06 17:40
收件人： apache/doris
抄送： ahboer; Author
主题： Re: [apache/doris] BE 1.2.4.1启动不了，报"Unrecognized VM option ':-CriticalJNINatives'，centos ,open jdk 1.8.0-372 (Discussion #19334)
CriticalJNiNatives is one of the startup option in the JVM, which usually defaults true. Due to Doris using some Java Native Interfaces, it needs to be turned off(-CriticalJNiNatives means turned off and +CriticalJNiNatives means turned on). In the start_be.sh script, there is a line like COMMON_OPTS="-Dsun.java.command=DorisBE -XX:-CriticalJNINatives". Generally, script execution will not cause problems, but according to your description, it seems that something wrong when you execute start_be.sh.
Perhaps you can try executing bash start_be.sh or change to a different version of JDK, such as OpenJDK Runtime Environment (build 1.8.0_372-b07)

**Comment by ihadoop** [comment]
i occur the same problem
